src/pages/SearchPage/index.js

```diff
@@ -0,0 +1,181 @@
+import PropTypes from 'prop-types';
+import React, {useEffect, useMemo, useState} from 'react';
+import {View} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import {usePersonalDetails} from '@components/OnyxProvider';
+import ScreenWrapper from '@components/ScreenWrapper';
+import SelectionList from '@components/SelectionList';
+import useDebouncedState from '@hooks/useDebouncedState';
+import useLocalize from '@hooks/useLocalize';
+import useNetwork from '@hooks/useNetwork';
+import useThemeStyles from '@hooks/useThemeStyles';
+import Navigation from '@libs/Navigation/Navigation';
+import * as OptionsListUtils from '@libs/OptionsListUtils';
+import Performance from '@libs/Performance';
+import * as ReportUtils from '@libs/ReportUtils';
+import reportPropTypes from '@pages/reportPropTypes';
+import * as Report from '@userActions/Report';
+import Timing from '@userActions/Timing';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import SearchPageFooter from './SearchPageFooter';
+
+const propTypes = {
+    /* Onyx Props */
+
+    /** Beta features list */
+    betas: PropTypes.arrayOf(PropTypes.string),
+
+    /** All reports shared with the user */
+    reports: PropTypes.objectOf(reportPropTypes),
+};
+
+const defaultProps = {
+    betas: [],
+    reports: {},
+};
+
+const searchRendered = () => {
+    Timing.end(CONST.TIMING.SEARCH_RENDER);
+    Performance.markEnd(CONST.TIMING.SEARCH_RENDER);
+};
+
+function SearchPage({betas, reports}) {
+    const [isScreenTransitionEnd, setIsScreenTransitionEnd] = useState(false);
+    const {translate} = useLocalize();
+    const {isOffline} = useNetwork();
+    const themeStyles = useThemeStyles();
+    const personalDetails = usePersonalDetails();
+
+    const offlineMessage = isOffline ? `${translate('common.youAppearToBeOffline')} ${translate('search.resultsAreLimited')}` : '';
+
+    const [searchValue, debouncedSearchValue, setSearchValue] = useDebouncedState('', 75);
```

Comment:
I think we should use this debounce time from CONST. It's 300 ms.
https://github.com/Expensify/App/blob/73f42a8b51bbcae45fa0f3f0b7e8bb8ca52c3576/src/CONST.ts#L713


discussion on it here - https://expensify.slack.com/archives/C05LX9D6E07/p1702907007858539?thread_ts=1702335974.199009&cid=C05LX9D6E07

Reply:
yeah, 75 is definitely to small! thanks for that!